src/backend/joanie/core/models/products.py

```diff
@@ -0,0 +1,296 @@
+            raise exceptions.InvalidCourseRuns(
+                f"{len(product_course_runs_available)} course runs have to be selected, "
+                f"{len(resource_links)} given"
+            )
+        # Make sure user chooses one of course runs for each necessary step (position)
+        # first extract valid course runs selected by user for each necessary step
+        valid_course_runs_selected = {}
+        for resource_link in resource_links:
+            # Course run has to be available for the product
+            # Raise ObjectDoesNotExist if it's not a resource_link of one of course runs
+            # available for the product
+            course_run = self.course_runs.get(resource_link=resource_link)
+            for position, course_runs in product_course_runs_available.items():
+                if course_run in list(course_runs):
+                    valid_course_runs_selected[position] = course_run
+        # We need exactly one course run selected for each necessary step
+        if len(valid_course_runs_selected) != len(product_course_runs_available):
+            raise exceptions.InvalidCourseRuns(
+                f"{len(product_course_runs_available)} course runs have to be selected, "
+                f"{len(valid_course_runs_selected)} given"
+            )
+
+        # So now everything is fine, we can create an order
+        order = Order.objects.create(product=self, owner=user)
+        for position in sorted(valid_course_runs_selected.keys()):
+            # associate each course run selected to the order
+            course_run = valid_course_runs_selected[position]
+            order.course_runs.add(course_run)
+            # then create enrollment for each course run
+            enrollment = Enrollment.objects.create(course_run=course_run, order=order)
+            # now we can enroll user to LMS course run
+            lms = LMSHandler.select_lms(course_run.resource_link)
+            # if no lms found we set enrollment and order to failure state
+            # this issue could be due to a bad setting or a bad resource_link filled,
+            # so we need to log this error to fix it quickly to joanie side
+            if lms is None:
+                order.state = enums.ORDER_STATE_FAILED
+                order.save()
+                enrollment.state = enums.ENROLLMENT_STATE_FAILED
+                enrollment.save()
+                logger.error(
+                    "No LMS configuration found for resource link: %s",
+                    course_run.resource_link,
+                )
+            # now enroll user to lms course run and change joanie enrollment state to 'in_progress'
+            else:
+                lms_enrollment = lms.set_enrollment(
+                    user.username, course_run.resource_link
+                )
+                if lms_enrollment["is_active"]:
+                    enrollment.state = enums.ENROLLMENT_STATE_IN_PROGRESS
+                    enrollment.save()
```

Comment:
I wonder if something is not missing here to support case where `set_enrollment` failed. In this case, we whould pass the enrollment and the order state to failed.